### PR TITLE
Use `instance-identity` as a plugin rather than a module

### PR DESCRIPTION
**Pull request by basil**
Not needed for BOM in the short term, but a generally positive maintenance step IMHO. CC @jglick

### CODE REVIEW

**Comment by basil** [comment]
Used my admin powers to replay the run with the `Jenkinsfile` changes: https://ci.jenkins.io/job/Plugins/job/github-plugin/job/PR-289/3/

File: pom.xml [hunk 1]
```diff
@@ -50,7 +50,7 @@
         <revision>1.34.6</revision>
         <changelist>-SNAPSHOT</changelist>
         <gitHubRepo>jenkinsci/github-plugin</gitHubRepo>
-        <jenkins.version>2.346.1</jenkins.version>
+        <jenkins.version>2.357</jenkins.version>
```
Comment (jglick) [inline pom.xml:53]: or prep for LTS
```suggestion
        <jenkins.version>2.361</jenkins.version>
```

**Review by jglick — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (basil) [reply pom.xml:53]: The stated purpose of this PR is to adapt to `instance-identity` changes, not to prep for LTS.

**Comment by basil** [comment]
CC @KostyaSha A merge and release of this PR would help us to plugin compatibility testing against this plugin after the `instance-identity` split in 2.357.

**Review by olamy — APPROVED** [review]

**Comment by basil** [comment]
@oleg-nenashev @KostyaSha Gentle ping. A merge and release of this plugin would facilitate PCT testing for the forthcoming 2.361.1 LTS release.

**Comment by basil** [comment]
Thanks for merging this PR, @KostyaSha! The [CI build is green](https://ci.jenkins.io/job/Plugins/job/github-plugin/job/master/). Are you interested in performing a release? The latter would help the core maintainers do plugin compatibility testing (PCT) in preparation for the forthcoming 2.361.1 LTS release.

**Pull request merged**